### PR TITLE
Added role pasing and authorization

### PR DESCRIPTION
**Pull request by KiLanger**
Currently, there is no buyer accessible enpoint, since @nk-coding and I discuessed, that there will be a seperate "isInStock" field.

https://frontend.gropius.duckdns.org/components/ba383e91-738a-488f-baf4-4b5d8876bd74/issues/0400015b-87f2-4258-a9f5-b1a6edcfe380

- [x]  Requirements of the issue are met
- [x]  Code has been reviewed
- [x]  Code is documented
- [x]  GraphQL related artefacts are documented

### CODE REVIEW

File: src/shared/enums/role.enum.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export enum Role {
+  SITE_ADMIN = 'site-admin',
```
Comment (delvh) [inline src/shared/enums/role.enum.ts:2]: That was/will be changed to `admin`

File: src/shared/guards/roles.guard.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    if (!requiredRoles) {
+      return true;
+    }
+
+    // get graphql context
+    const ctx = GqlExecutionContext.create(context);
+    const { headers } = ctx.getContext().request;
+
+    // check if user is authorized by gateway
+    if(!headers['authorized-user']) {
```
Comment (delvh) [inline src/shared/guards/roles.guard.ts:27]: Erm... Isn't the header called 
```suggestion
    if(!headers['Authorized-User']) {
```

File: src/shared/guards/roles.guard.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      return false;
+    }
+
+    // check if user has required roles
+    const roles = JSON.parse(headers['authorized-user']).roles;
```
Comment (delvh) [inline src/shared/guards/roles.guard.ts:32]:
```suggestion
    const roles = JSON.parse(headers['Authorized-User']).roles;
```

File: src/inventory/inventory.resolver.ts [hunk 4]
```diff
@@ -164,6 +171,7 @@ export class InventoryResolver {
     );
   }
 
+  @Roles(Role.SITE_ADMIN)
```
[the remark this answers is not in the document]
Comment (KiLanger) [reply src/inventory/inventory.resolver.ts:174]: Product Items have a state, they should probably not be deleted. If the customer received the product, he can still return it later.
This is meant as a safeguard for user-errors but should probably be not availabe for employees. (IE if you screw up, go talk to your supervisor so you can not disappear 10 IPhones)

File: src/shared/enums/role.enum.ts (hunk 1 above)
Comment (KiLanger) [reply src/shared/enums/role.enum.ts:2]: Ok, when?

File: src/shared/guards/roles.guard.ts (hunk 2 above)
Comment (KiLanger) [reply src/shared/guards/roles.guard.ts:27]: Its lowercased by the body parser

File: src/shared/enums/role.enum.ts (hunk 1 above)
Comment (nk-coding) [reply src/shared/enums/role.enum.ts:2]: yesterday or so

Comment (delvh) [reply src/shared/enums/role.enum.ts:2]: https://github.com/MiSArch/keycloak/pull/9

Comment (delvh) [reply src/shared/enums/role.enum.ts:2]:
```suggestion
  SITE_ADMIN = 'admin',
```

**Review by delvh — APPROVED** [review]

**Review by nk-coding — APPROVED** [review]
works

**Pull request merged**